es55x/src/main/scala/tech/beshu/ror/es/ResponseFieldsFiltering.scala

```diff
@@ -0,0 +1,54 @@
+package tech.beshu.ror.es
+
+import org.elasticsearch.common.xcontent.cbor.CborXContent
+import org.elasticsearch.common.xcontent.json.JsonXContent
+import org.elasticsearch.common.xcontent.smile.SmileXContent
+import org.elasticsearch.common.xcontent.yaml.YamlXContent
+import org.elasticsearch.common.xcontent.{NamedXContentRegistry, XContentBuilder, XContentType}
+import org.elasticsearch.rest.{BytesRestResponse, RestResponse}
+import tech.beshu.ror.accesscontrol.domain.ResponseFieldsFiltering.{AccessMode, ResponseFieldsRestrictions}
+
+import scala.collection.JavaConverters._
+
+trait ResponseFieldsFiltering {
+
+  private var responseFieldsRestrictions: Option[ResponseFieldsRestrictions] = None
```

Comment:
I don't think making this variable atomic would change much because `setResponseFieldRestrictions` method is only setting it without reading it first and `filterRestResponse` is performing only single read.

Reply:
personally I don't like to use @volatile on JVM. Atomic var [has the same semantics](https://javarevisited.blogspot.com/2020/04/difference-between-atomic-volatile-and-synchronized-in-java-multi-threading.html) but the code can be easily extended, so I don't see a reason why we should not use them by default.